elephant/statistics.py

```diff
@@ -799,23 +809,416 @@ def complexity_pdf(spiketrains, binsize):
+                          'the affected spikes into the following bin. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.'.format(num_rounding_corrections))
+        start_bins[rounding_error_indices] += .5
+
+        start_bins = start_bins.astype(int)
+
+        rounding_error_indices = conv._detect_rounding_errors(stop_bins,
+                                                              self.tolerance)
+
+        num_rounding_corrections = rounding_error_indices.sum()
+        if num_rounding_corrections > 0:
+            warnings.warn('Correcting {} rounding errors by shifting '
+                          'the affected spikes into the following bin. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.'.format(num_rounding_corrections))
+        stop_bins[rounding_error_indices] += .5
+
+        stop_bins = stop_bins.astype(int)
+
+        for idx, (start, stop) in enumerate(zip(start_bins, stop_bins)):
+            time_hist[start:stop] = \
+                    self.epoch.array_annotations['complexity'][idx]
+
+        time_hist = neo.AnalogSignal(
+            signal=time_hist.reshape(time_hist.size, 1),
+            sampling_period=self.bin_size, units=pq.dimensionless,
+            t_start=self.t_start)
+
+        empty_bins = (self.t_stop - self.t_start - self.epoch.durations.sum())
+        empty_bins = empty_bins.rescale(self.bin_size.units) / self.bin_size
+        if conv._detect_rounding_errors(empty_bins, tolerance=self.tolerance):
+            warnings.warn('Correcting a rounding error in the histogram '
+                          'calculation by increasing num_bins by 1. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.')
+            empty_bins += 1
+        empty_bins = int(empty_bins)
+
+        complexity_hist[0] = empty_bins
+
+        return time_hist, complexity_hist
+
+    def _epoch_no_spread(self):
+        """
+        Get an epoch object of the complexity distribution with `spread` = 0
+        """
+        left_edges = self.time_histogram.times
+        durations = self.bin_size * np.ones(self.time_histogram.shape)
+        if self.sampling_rate:
+            # ensure that spikes are not on the bin edges
+            bin_shift = .5 / self.sampling_rate
+            left_edges -= bin_shift
+        else:
+            warnings.warn('No sampling rate specified. '
+                          'Note that using the complexity epoch to get '
+                          'precise spike times can lead to rounding errors.')
+
+        # ensure that an epoch does not start before the minimum t_start
+        min_t_start = min([st.t_start for st in self.input_spiketrains])
+        if left_edges[0] < min_t_start:
+            left_edges[0] = min_t_start
+            durations[0] -= bin_shift
```

Comment:
`bin_shift` is not defined when self.sampling_rate if False.

Reply:
True, I fixed it.